Bug/mc 11900 hypercloud mandatory cluster

m-orfanos opened the pull request:
### Fixes [MC-11900](https://cloud-ops.atlassian.net/browse/MC-11900)

#### Changes made
The kubernetes documentation for the hypercloud did not mention the `cluster id` mandatory field for some of the entities. It was added.

m-orfanos committed:
Adds k8s config map doc to hyperclouds

m-orfanos committed:
Adds doc for mandatory cluster id in hyperclouds

m-orfanos commented:
@jasondias9 @y0geshchaudhary @Gerudhoh FYI the hypercloud plugin APIs were tested for all the entities affected in this PR. We good 👍

Gerudhoh commented on source/includes/kubernetes_extension/_k8_namespaces.md:
```diff
@@ -1,106 +1,70 @@
 curl -X GET \
    -H "MC-Api-Key: your_api_key" \
-   "https://cloudmc_endpoint/v1/services/k8s/an_environment/namespaces"
+   "https://cloudmc_endpoint/v1/services/a_service/an_environment/namespaces?cluster_id=a_cluster_id"
```
good touch changing "k8s" to "a service"

y0geshdev commented on source/includes/kubernetes_extension/_k8_configmaps.md:
```diff
@@ -7,45 +7,59 @@
 
 Retrieve a list of all config maps in a given [environment](#administration-environments).
 
+| Required                   | &nbsp;                                                  |
```
Should we add a section like this?
It just looks a bit odd IMO. How do we feel about putting the same content in a 👇 way:
> ***Required<br/>*** `cluster_id`:   The id of the cluster in which to list the config maps.

m-orfanos replied:
This section matches what's already in pods, released and charts (we need to update those docs as well if we want to change it). It also matches the attributes listing right below.

Gerudhoh commented on source/includes/kubernetes_extension/_k8_statefulsets.md:
```diff
@@ -128,16 +133,20 @@ curl -X GET \
 | `metadata` <br/>*object*                   | The metadata of the stateful set                                |
 | `metadata.creationTimestamp` <br/>*string* | The date of creation of the stateful set as a string            |
-| `metadata.labels` <br/>*map*               | The labels associated to the stateful                           |
+| `metadata.labels` <br/>*map*               | The labels associated to the stateful set                       |
```
good catch!

Gerudhoh commented on source/includes/kubernetes_extension/_k8_deployments.md:
```diff
@@ -7,67 +7,280 @@
-      "metadata": {
-        "annotations": {
-          "deployment.kubernetes.io/revision": "1"
+    "data": [
```
is all this data really necessary to display?

m-orfanos replied:
I could change all the nested objects to be an empty object 

```
metadata: { ... },
spec: {...},
etc.
```

Gerudhoh replied:
I think that's a good call! Will reduce clutter + noise

m-orfanos replied:
Done  363ce1f7606be98ce0bf0a53a32895438f13fd0f

jasondias9 reviewed (commented):
As per the new PR template - you should use the `js` formatter instead of `json`. I think some of these need changing

y0geshdev reviewed (approved):
LGTM

m-orfanos committed:
Removes nested objects from response examples

The full examples of responses were extremely long and cluttered the
documentation. Nested objects were replaced with an empty object.

The users can always go look at the official Kubernetes documentation
to get examples since they're (nearly) identical.

Gerudhoh reviewed (approved):
LGTM!
